apps/dashboard/app/lib/smart_attributes/attributes/auto_scripts.rb

```diff
@@ -26,6 +27,20 @@ def self.script_options_from_directory(dir)
         [File.basename(file), file]
       end.sort_by(&:first)
     end
+
+    def self.default_script_value(initial_opts, script_opts)
+      return nil if !initial_opts[:value] || script_opts.empty?
+
+      # Replace directory if the script is present in correct directory, otherwise delete value
+      valid_opt = script_opts.select { |opt| opt.first == File.basename(initial_opts[:value]) }.first
+      if valid_opt
+        initial_opts[:value] = valid_opt.last
+      elsif script_opts&.none? { |opt| opt.include?(initial_opts[:value]) }
+        initial_opts.delete(:value)
+      end
+
+      (initial_opts[:value] || script_opts.first.last).to_s
+    end
```

Comment:
I'm struggling to understand this. I think because it does 2 things - potentially modifies `initial_opts` _and_ has a return value.

Can we restructure this so it just has the return value with no modification to the things passed into it?

When I was investigating the tests, I came up with this - I also think a simple name change (and type change of the first argument) helped me figure out what it's attempting to do.
```ruby
# called through  default_value(opts[:default], options)

    def self.default_value(default, scripts)
      if scripts.empty?
        nil
      elsif default.nil? || default.to_s.empty?
        scripts.first[1]
      elsif scripts.map { |s| s[1] }.include?(default)
        default
      else
        scripts.first[1]
      end
    end
```